pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -531,17 +536,36 @@ func (m *migrationMonitor) processInflightMigration(dom cli.VirDomain, stats *li
 		// If we were to abort the migration due to a timeout while in post copy,
 		// then it would result in that active state being lost.
 
-	case m.shouldTriggerPostCopy(elapsed):
-		logger.Info("Starting post copy mode for migration")
-		// if a migration has stalled too long, post copy will be
-		// triggered when allowPostCopy is enabled
-		err := dom.MigrateStartPostCopy(uint32(0))
-		if err != nil {
-			logger.Reason(err).Error("failed to start post migration")
-			return nil
-		}
+	case m.shouldAssistMigrationToComplete(elapsed) && !m.isPausedMigration():
+
+		if m.options.AllowPostCopy {
+			logger.Info("Starting post copy mode for migration")
+			// if a migration has stalled too long, post copy will be
+			// triggered when allowPostCopy is enabled
+			err := dom.MigrateStartPostCopy(uint32(0))
+			if err != nil {
+				logger.Reason(err).Error("failed to start post migration")
+				return nil
+			}
+			m.l.updateVMIMigrationMode(v1.MigrationPostCopy)
+		} else {
+
+			logger.Info("Pausing the guest to allow migration to complete")
+			// if a migration has stalled too long, the guest will be paused
+			// to complete the migration when allowPostCopy is disabled
+			err := dom.Suspend()
+			if err != nil {
+				logger.Reason(err).Error("Signalling suspension failed.")
+				return nil
+			}
+			logger.Infof("Signaled pause for %s", m.vmi.GetObjectMeta().GetName())
 
-		m.l.updateVMIMigrationMode(v1.MigrationPostCopy)
+			// update acceptableCompletionTime to prevent premature migration
+			// cancellation
+			m.acceptableCompletionTime *= 2
```

Comment:
well, no, since this function controls the guest activity. When the guest is paused, no memory is being polluted hence the VM will migrate faster.
However, this shouldn't affect the progressTimeout because it means: "What is the time we should wait until the migrations make progress - pages are being copied". 
Even if the VM is paused, pages are still being copied between the source and the target.

Reply:
Why do we double this?

Reply:
It just gives us enough buffer.

Reply:
My logic was that initially, we assessed the guest would complete the migration in `acceptableCompletionTime`. Now that time passed and paused. The guest surely should complete in the original `acceptableCompletionTime` now that it's not active.